pkg/github/server.go

```diff
@@ -212,9 +247,85 @@ func OptionalPaginationParams(r mcp.CallToolRequest) (PaginationParams, error) {
 	if err != nil {
 		return PaginationParams{}, err
 	}
+	after, err := OptionalParam[string](r, "after")
+	if err != nil {
+		return PaginationParams{}, err
+	}
 	return PaginationParams{
-		page:    page,
-		perPage: perPage,
+		Page:    page,
+		PerPage: perPage,
+		After:   after,
+	}, nil
+}
+
+// OptionalCursorPaginationParams returns the "perPage" and "after" parameters from the request,
+// without the "page" parameter, suitable for cursor-based pagination only.
+func OptionalCursorPaginationParams(r mcp.CallToolRequest) (CursorPaginationParams, error) {
+	perPage, err := OptionalIntParamWithDefault(r, "perPage", 30)
+	if err != nil {
+		return CursorPaginationParams{}, err
+	}
+	after, err := OptionalParam[string](r, "after")
+	if err != nil {
+		return CursorPaginationParams{}, err
+	}
+	return CursorPaginationParams{
+		PerPage: perPage,
+		After:   after,
+	}, nil
+}
+
+type CursorPaginationParams struct {
+	PerPage int
+	After   string
+}
+
+// ToGraphQLParams converts cursor pagination parameters to GraphQL-specific parameters.
+func (p CursorPaginationParams) ToGraphQLParams() (GraphQLPaginationParams, error) {
+	if p.PerPage > 100 {
+		return GraphQLPaginationParams{}, fmt.Errorf("perPage value %d exceeds maximum of 100", p.PerPage)
+	}
+	if p.PerPage < 0 {
+		return GraphQLPaginationParams{}, fmt.Errorf("perPage value %d cannot be negative", p.PerPage)
+	}
+	first := int32(p.PerPage)
+
+	var after *string
+	if p.After != "" {
+		after = &p.After
+	}
+
+	return GraphQLPaginationParams{
+		First: &first,
+		After: after,
+	}, nil
+}
+
+type GraphQLPaginationParams struct {
+	First *int32
+	After *string
+}
+
+// ToGraphQLParams converts REST API pagination parameters to GraphQL-specific parameters.
+// This converts page/perPage to first parameter for GraphQL queries.
+// If After is provided, it takes precedence over page-based pagination.
+func (p PaginationParams) ToGraphQLParams() (GraphQLPaginationParams, error) {
```

Comment:
What do you think of returning a pointer to the`GraphQLPaginationParams`? 
Then instead of returning an empty object we could return nil to indicate failure:
```
return nil, fmt.Errorf("perPage value %d exceeds maximum of 100", p.PerPage) 
```